internal/cmd/cmd.go

```diff
@@ -281,25 +333,50 @@ func (c *Cmd) Wait() error {
 				// Close the process to cancel any reads to stdout or stderr.
 				c.Process.Close()
 				if c.Log != nil {
-					c.Log.Warn("timed out waiting for stdio relay")
+					c.Log.WithField(logfields.Timeout, c.CopyAfterExitTimeout.String()).
+						Warn("timed out waiting for stdio relay")
 				}
 			}
 		}()
 	}
+
 	ioErr := c.iogrp.Wait()
-	if ioErr == nil {
-		ioErr, _ = c.stdinErr.Load().(error)
+	if inErr, _ := c.stdinErr.Load().(error); inErr != nil {
+		if ioErr == nil {
+			ioErr = inErr
+		} else {
+			// cannot wrap two errors at once, so one will be wrapped via `%v`
+			ioErr = fmt.Errorf("multiple IO copy errors: %v; %w", inErr, ioErr)
+		}
+	}
+	if err == nil {
+		err = ioErr
+	} else {
+		// wrap (prioritize) IO errors over `afterExit` errors
+		err = fmt.Errorf("io error: %w; other error: %v", ioErr, err)
 	}
+	// close the channel first, to prevent `Process.Kill` being called mid-`Process.Close`
 	close(c.allDoneCh)
-	c.Process.Close()
+	// Process could have been closed by IO timeout handling
+	if cerr := c.Process.Close(); cerr != nil && !errors.Is(cerr, hcs.ErrAlreadyClosed) {
```

Comment:
why not use `isIOChannelClosedErr` call here?

Reply:
I didnt want to ignore IO close errors, since its only looking if the process was previously closed, and not if an error arose while closing.